plugins/funind/glob_term_to_relation.ml

```diff
@@ -1275,17 +1275,20 @@ let do_build_inductive
        let evd,t = Typing.type_of env evd (mkConstU c) in
        evd,
        Environ.push_named (LocalAssum (id,t))
-			   (* try *)
-			   (*   Typing.e_type_of env evd (mkConstU c) *)
-			   (* with Not_found -> *)
-			   (*   raise (UserError("do_build_inductive", str "Cannot handle partial fixpoint")) *)
 			 env
       )
       funnames
       (Array.of_list funconstants)
       (evd,Global.env ())
   in
-  let resa = Array.map (build_entry_lc env  funnames_as_set []) rta in
+  (* we solve and replace the implicits *)
+  let rta =
+    Array.mapi (fun i rt ->
+        let _,t = Typing.type_of env evd (mkConstU ((Array.of_list funconstants).(i))) in
```

Comment:
While looking for the change to do I just noticed that  `evd` is already the result of typing all the constants in functonstants (just 9 lines above). We can then safely dump the evar_map IMHO.